Documentation/contributing/development/introducing_new_crds.rst

```diff
@@ -0,0 +1,404 @@
+   +// called on agent startup but is idempotent and safe to call again.
+   +func createBGPPoolCRD(clientset apiextensionsclient.Interface) error {
+   +       ciliumCRD := GetPregeneratedCRD(BGPPoolCRDName)
+   +
+   +       return createUpdateCRD(
+   +               clientset,
+   +               BGPPoolCRDName,
+   +               constructV1CRD(k8sconstv2alpha1.BGPPName, ciliumCRD),
+   +               newDefaultPoller(),
+   +       )
+   +}
+   +
+    // createUpdateCRD ensures the CRD object is installed into the K8s cluster. It
+    // will create or update the CRD and its validation schema as necessary. This
+    // function only accepts v1 CRD objects, and defers to its v1beta1 variant if
+
+Getting Your CRDs Installed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Your new CRDs must be installed into Kubernetes. This is controlled in
+the ``pkg/k8s/synced/crd.go`` file.
+
+Here is an example diff which installs the CRDs ``v2alpha1.BGPPName``
+and ``v2alpha.BGPPoolName``:
+
+.. code-block:: diff
+
+   diff --git a/pkg/k8s/synced/crd.go b/pkg/k8s/synced/crd.go
+   index 52d975c449..10c554cf8a 100644
+   --- a/pkg/k8s/synced/crd.go
+   +++ b/pkg/k8s/synced/crd.go
+   @@ -42,6 +42,11 @@ func agentCRDResourceNames() []string {
+                   CRDResourceName(v2.CCNPName),
+                   CRDResourceName(v2.CNName),
+                   CRDResourceName(v2.CIDName),
+   +               CRDResourceName(v2.CIDName),
+   +               // TODO(louis) make this a conditional install
+   +               // based on --enable-bgp-control-plane flag
+   +               CRDResourceName(v2alpha1.BGPPName),
+   +               CRDResourceName(v2alpha1.BGPPoolName),
+           }
+
+Updating RBAC Roles
+~~~~~~~~~~~~~~~~~~~
+
+Cilium is installed with a service account and this service account
+should be given RBAC permissions to access your new CRDs. The following
+files should be updated to include permissions to create, read, update, and delete 
+your new CRD.
+
+.. code-block:: text
+
+   install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   install/kubernetes/cilium/templates/cilium-operator/clusterrole.yaml
+   install/kubernetes/cilium/templates/cilium-preflight/clusterrole.yaml
+
+Here is a diff of updating the Agent's cluster role template to include
+our new BGP CRDs:
+
+.. code-block:: diff
+
+   diff --git a/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml b/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   index 9878401a81..5ba6c30cd7 100644
+   --- a/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   +++ b/install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
+   @@ -102,6 +102,8 @@ rules:
+      - ciliumlocalredirectpolicies/finalizers
+      - ciliumegressnatpolicies
+      - ciliumendpointslices
+   +  - ciliumbgppeeringpolicies
+   +  - ciliumbgploadbalancerippools
+      verbs:
+      - '*'
```

Comment:
We should add a note that the "verbs" section should be explicit about which methods/verbs are actually required.

Reply:
Ah, @aanm somehow I missed this comment. Adding this now.